src/Components/ResourcePicker/ResourcePicker.tsx

```diff
@@ -547,47 +514,122 @@ const ResourcePicker: React.FC<IResourcePickerProps> = ({
                         {error.name}
                     </MessageBar>
                 )}
-                {resourcesState.isLoading && (
-                    <Spinner
-                        size={SpinnerSize.xSmall}
-                        label={loadingLabel ?? loadingLabelText}
-                        ariaLive="assertive"
-                        labelPosition="right"
-                    />
-                )}
             </div>
         ),
-        [
-            error,
-            classNames,
-            resourcesState,
-            label,
-            loadingLabel,
-            loadingLabelText,
-            t
-        ]
+        [error, classNames, resourcesState, label, loadingLabel, t]
     );
 
+    const CustomMenuList = (props) => {
+        const OPTION_HEIGHT = 32;
+        const { children, maxHeight } = props;
+
+        return options.length <= 1 ? (
+            <components.MenuList {...props} />
+        ) : (
+            <FixedSizeList
+                height={maxHeight}
+                itemCount={children.length}
+                itemSize={OPTION_HEIGHT}
+                className={classNames.menuList}
+            >
+                {({ index, style }) => (
+                    <div style={style}>{children[index]}</div>
+                )}
+            </FixedSizeList>
+        );
+    };
+
+    const CustomOption = (props) => {
+        if (props.data.type === 'header') {
```

Comment:
Why do you prefer to create header options instead of using react-select groups?

Reply:
yeah, I tried it there was a whole trial-error about using select groups. The reason I didn't use it was because I was not able to render each item correctly in a virtualized list when I use react-window as it expects a one-dimensional flat list of options. Since each children in DOM has might have sub-children (in react-select group there are sub options field), the rendered option texts were overlapping.